app/graphql/types/data_management_plan_type.rb

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+class DataManagementPlanType < BaseObject
+  implements DoiItem
+
```

Comment:
What about adding a  `datasets` field that aggregates DoiItem **citations** and **references** together but filters them by type "**Dataset**". This would remove significant processing for the user of the DMP which would be looking for datasets related to the DMP and not DOIs related. That aligns, IMO, closer to the vision of the DMP.

Reply:
If possible I would avoid any functionality that is specific to DMPs. It makes it more complicated for users to use the API, and I assume we need to refactor when we release the next schema with support for ResourceTypeGeneral `dataManagementPlan`.

What one can do instead to customize the query is using an [inline fragment](https://graphql.org/learn/schema/):

```
query HeroForEpisode($ep: Episode!) {
  hero(episode: $ep) {
    name
    ... on Droid {
      primaryFunction
    }
  }
}
```

Reply:
> It makes it more complicated for users to use the API,

Can you elaborate on this? I do not think I understand why adding functionality to a more specific type would make this more complicated for users. Particularly when they will have to scan the response to find what they actually need. The use case of the DMPs talks about, exclusively,  `datasets`. It feels to me that without adding this functionality in the type, we are making the user to work more and to have knowledge about what constitute a `datasets` from all the metadata response. 

I would like to understand why is this more complicated.

Reply:
The design of the DataCite GraphQL API tries to be as generic as possible, e.g. in terms of supported fields. Assuming that users will not only query for DMPs, or at least they may reuse pre-existing notebooks and tweak them. Please look into the suggestions I made regarding the query on the client side.